### DISCUSSION TITLE
¿Como puedo enviar una  imagen con texto?

### DISCUSSION THREAD

**Discussion by luisucan** [codigoencasa/builderbot#1013] [Q&A]
dentro del flujo no encuentro la forma correcta de enviar una imagen con texto y que este espere a que se haya enviado el mensaje para luego enviar otro tipo de mensaje

**Comment by JeanCaicedo** [comment]
Puedes enviar la imagen con el texto como caption usando la opcion `media` de `addAnswer`:

```js
const flow = addKeyword('hola')
  .addAnswer('Este texto acompana la imagen', {
    media: 'https://ejemplo.com/imagen.png'
  })
  .addAnswer('Este mensaje se envia despues de la imagen')
```

Los `addAnswer` encadenados se envian en orden, asi que el segundo mensaje espera a que la imagen se haya enviado.

Si necesitas hacerlo dinamicamente dentro del flujo, usa `flowDynamic` con `await`, que garantiza que el mensaje se envio antes de continuar:

```js
.addAction(async (ctx, { flowDynamic }) => {
  await flowDynamic([{ body: 'Texto con la imagen', media: 'https://ejemplo.com/imagen.png' }])
  await flowDynamic('Mensaje siguiente')
})
```
